### PR TITLE
Add delay to new slime spawns after slime death

### LINKED ISSUE #5337: When killed, slimes split too soon
Client version: 1.12.2
Server OS: Windows 11
Cuberite Commit id: 95bc2eac222a12c049b1aef799ac280fa9123af4

### Expected behavior
Slimes split after the end of their dying animation, like squids drop after their death.

### Actual behavior
The moment slimes' hp reach 0, they split.

### Steps to reproduce the behavior
Kill slimes.

### PR DESCRIPTION
**Pull request by KingCol13**
Fixes #5337

I couldn't find the delay documented anywhere so I guessed 20 ticks. It looks about right.

### CODE REVIEW

**Review by 12xx12 — APPROVED** [review]
One second sounds about right when choosing a value for this.

File: src/Mobs/Slime.cpp [hunk 1]
```diff
@@ -79,10 +80,18 @@ void cSlime::KilledBy(TakeDamageInfo & a_TDI)
...
+			int Size = m_Size/2;
+			auto lambda = [SpawnPos, Yaw, Size](cWorld &a_World)
+			{
+				auto NewSlime = std::make_unique<cSlime>(Size);
+				NewSlime->SetPosition(SpawnPos);
+				NewSlime->SetYaw(Yaw);
+				a_World.SpawnMobFinalize(std::move(NewSlime));
+			};
+			cTickTime DelayTime(20);
+			m_World->ScheduleTask(DelayTime, lambda);
```
Comment (tigerw) [inline src/Mobs/Slime.cpp:94]: Just a bit of style, haven't actually compiled this to check.

```suggestion
			// Queue slimes to be spawned after the 1 second death animation finished playing:
			m_World->ScheduleTask(cTickTime(20), [
				Position = GetPosition() + Vector3d(AddX, 0.5, AddZ),
				Yaw = Random.RandReal(360.0f),
				Size = m_Size / 2
			](cWorld & a_World)
			{
				auto NewSlime = std::make_unique<cSlime>(Size);
				NewSlime->SetPosition(Position);
				NewSlime->SetYaw(Yaw);
				a_World.SpawnMobFinalize(std::move(NewSlime));
			});
```

**Comment by tigerw** [comment]
The death animation is indeed 20 ticks:
https://github.com/cuberite/cuberite/blob/600dfbd8726409abc19e98ac99cca5746cbf7ebe/src/Mobs/Monster.cpp#L281-L287

File: src/Mobs/Slime.cpp (hunk 1 above)
Comment (KingCol13) [reply src/Mobs/Slime.cpp]: Cheers, any kind of suggestion very welcome. Can confirm it does compile :)

**Review by bearbin — APPROVED** [review]

**Pull request merged**